docs/configuration/schema-config-reference.md

```diff
@@ -0,0 +1,104 @@
+
+You can configure the schemas using a YAML config file, that you can point to using the `-schema-config-file` flag. It has the following YAML spec:
+
+```yaml
+configs: []<period_config>
+```
+
+Where `period_config` is
+```
+from: <Date>
+store: <string> // The index client to use.
+object_store: <string> // The object client to use. If none is specified, `store` is used for storing chunks as well.
+schema: <string> // The schema version to use. Valid ones are v1, v2, v3,... v6, v9, v10, v11. Recommended for production: v9
+index: <periodic_table_config>
+chunks: <periodic_table_config>
+```
+
+Where `periodic_table_config` is
+```
+prefix: <string> // The prefix to use for the tables.
+period: <time.Duration> // We typically run Cortex with new tables every week to keep the index size low and to make retention easier. This sets the period at which new tables are created and used. Typically 168h (1week).
+tags: <map[string]string> // The tags that can be set on the dynamo table. 
+```
+
+Now an example of this file (also something recommended when starting out) is:
+```
+configs:
+  - from: "2020-03-01" // Or typically a week before the Cortex cluster was created.
+    schema: v9
+    index:
+      period: 168h
+      prefix: cortex_index_
+    chunks:                     // Chunks section is optional and required only if you're not using a separate object store.
+      period: 168h
+      prefix: cortex_chunks  
+    store: aws-dynamo/bigtable-hashed/cassandra/boltdb
+    object_store: <above options>/s3/gcs/azure/filesystem
+```
+
+An example of an advanced schema file with a lot of changes:
+```
+configs:
+  - from: "2018-08-23"
+    schema: v9
+    chunks:
+        period: 168h0m0s
+        prefix: dev_chunks_
+    index:
+        period: 168h0m0s
+        prefix: dev_index_
+    store: gcp-columnkey
+  - from: "2019-02-13"
+    schema: v9
+    chunks:
+        period: 168h
+        prefix: dev_chunks_
+    index:
+        period: 168h
+        prefix: dev_index_
+    object_store: gcs
+    store: gcp-columnkey
+  - from: "2019-02-24"
+    schema: v9
+    chunks:
+        period: 168h
+        prefix: dev_chunks_
+    index:
+        period: 168h
+        prefix: dev_index_
+    object_store: gcs
+    store: bigtable-hashed
+  - from: "2019-03-05"
+    schema: v10
+    chunks:
+        period: 168h
+        prefix: dev_chunks_
+    index:
+        period: 168h
+        prefix: dev_index_
+    object_store: gcs
+    store: bigtable-hashed
+```
+
+Note how we started out with v9 and just Bigtable, but later added GCS as the object store, finally moving to v10. This is a complex schema file from out dev cluster that saw a lot of changes dating all the way back to 2018, a typical schema config file usually has one or two schema versions.
+
+### Migrating from flags to schema file
+
+We've recently removed the ability to configure schema via flags. If you were using flags, you need to port your config from flags to the config file.
```

Comment:
I would suggest to never use time-sensitive words like "recently" in the doc, because you never know when this doc will be updated again and the "recently" may easily become months or years.

```suggestion
Legacy versions of Cortex did support the ability to configure schema via flags. If you are still using flags, you need to migrate your configuration from flags to the config file.
```